test/ex_machina/ecto_test.ex

```diff
@@ -0,0 +1,77 @@
+defmodule ExMachina.EctoTest do
+  use ExUnit.Case, async: true
+
+  defmodule TestRepo do
+    def insert!(record) do
+      send self, {:created, record}
+      record
+    end
+  end
+
+  defmodule MyApp.Book do
+    defstruct title: nil, publisher: nil, __meta__: %{__struct__: Ecto.Schema.Metadata}, publisher_id: 1
+
+    def __schema__(:associations) do
+      [:publisher]
+    end
+  end
+
+  defmodule MyApp.ExMachina do
+    use ExMachina.Ecto, repo: TestRepo
+
+    def factory(:book) do
+      %MyApp.Book{
+        title: "Foo"
+      }
+    end
+
+    def factory(:user) do
+      %{
+        id: 3,
+        name: "John Doe",
+        admin: false
+      }
+    end
+  end
+
+  test "fields_for/2 removes Ecto specific fields" do
+    assert MyApp.ExMachina.fields_for(:book) == %{
+      title: "Foo",
+      publisher_id: 1,
+    }
+  end
+
+  test "fields_for/2 raises when passed a map" do
+    assert_raise ArgumentError, fn ->
+      MyApp.ExMachina.fields_for(:user)
+    end
+  end
+
+  test "create/2 builds the factory and saves it in the repo" do
+    user = MyApp.ExMachina.create(:user, admin: true)
+
+    assert user == %{id: 3, name: "John Doe", admin: true}
+    assert_received {:created, %{name: "John Doe", admin: true}}
+  end
+
+  test "create_pair/2 creates the factory and saves it 2 times" do
+    users = MyApp.ExMachina.create_pair(:user, admin: true)
+
+    created_user = %{id: 3, name: "John Doe", admin: true}
+    assert users == [created_user, created_user]
+    assert_received {:created, ^created_user}
+    assert_received {:created, ^created_user}
+    refute_received {:created, _}
+  end
+
+  test "create_list/3 creates factory and saves it passed in number of times" do
+    users = MyApp.ExMachina.create_list(3, :user, admin: true)
+
+    created_user = %{id: 3, name: "John Doe", admin: true}
+    assert users == [created_user, created_user, created_user]
+    assert_received {:created, ^created_user}
+    assert_received {:created, ^created_user}
+    assert_received {:created, ^created_user}
+    refute_received {:created, _}
+  end
```

Comment:
I think we could remove all of the `create...` tests and instead call `save_record` and ensure that it saves. This would simplify this test file a lot I think. 

create, create_list, etc. are tested in the other file already